### PR TITLE
A3142 runtime descriptor endpoints

### PR DESCRIPTION
**Pull request by dmitrys-odysseus**

### CODE REVIEW

File: src/main/java/com/odysseusinc/arachne/datanode/scheduler/CentralScheduler.java [hunk 1]
```diff
@@ -81,10 +83,10 @@ private void checkModeSwitching() {
         if (currentDataNode != null  && !userService.findStandaloneUsers().isEmpty()) {
             centralIntegrationService.relinkUsersToDataNodeOnCentral(currentDataNode, userService.findStandaloneUsers());
         }
-        List<DataSource> dataSources = dataSourceService.findStandaloneSources();
-        if (dataSources.size() > 0) {
-            throw new BeanInitializationException("Cannot switch mode from Standalone to Network - there are some data sources not linked to the Central.");
-        }
+//        List<DataSource> dataSources = dataSourceService.findStandaloneSources();
```
Comment (wivern) [inline src/main/java/com/odysseusinc/arachne/datanode/scheduler/CentralScheduler.java]: Why this change?

File: src/main/java/com/odysseusinc/arachne/datanode/environment/EnvironmentDescriptorService.java [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+    }
+
+    @Transactional
+    public List<EnvironmentDescriptorDto> listActive() {
+        return getActive().map(entity ->
+                EnvironmentDescriptorDto.of(entity.getId(), entity.getDescriptorId(), entity.getLabel(), entity.getJson())
+        ).collect(Collectors.toList());
+    }
+
+    private Stream<EnvironmentDescriptor> getActive() {
```
Comment (wivern) [inline src/main/java/com/odysseusinc/arachne/datanode/environment/EnvironmentDescriptorService.java:93]: Why not to use query condition to filter non-actives?

File: src/main/java/com/odysseusinc/arachne/datanode/environment/EnvironmentDescriptorService.java [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+        return getAll().filter(descriptor ->
+                descriptor.getTerminated() == null
+        );
+    }
+
+    private Stream<EnvironmentDescriptor> getAll() {
```
Comment (wivern) [inline src/main/java/com/odysseusinc/arachne/datanode/environment/EnvironmentDescriptorService.java:99]: Why it's not used spring JPA repository like in others?

File: src/main/java/com/odysseusinc/arachne/datanode/util/JpaSugar.java [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+import javax.persistence.criteria.CriteriaQuery;
+import javax.persistence.criteria.CriteriaUpdate;
+import javax.persistence.criteria.Path;
+import javax.persistence.criteria.Root;
+import org.springframework.data.domain.Pageable;
+
+/**
+ * Syntactic sugar to get more expressive semantics on the JPA operations.
+ */
+public interface JpaSugar {
```
Comment (wivern) [inline src/main/java/com/odysseusinc/arachne/datanode/util/JpaSugar.java]: Do we really need this here?

**Review by alex-odysseus — DISMISSED** [review]

**Review by alex-odysseus — DISMISSED** [review]

**Review by alex-odysseus — APPROVED** [review]
